### DISCUSSION TITLE
payout: contract changes discussions

### DISCUSSION THREAD

**Discussion by thelostone-mc** [gitcoinco/grants-stack#1128] [Q&A]
### Description

This is to be picked up after we've all the previous steps are wrapped and we have a deployable end to end flow.

**RoundFactory**
- Whitelist contract -> for curation / approved payout flavours/voting flavours? Open to discussion but if we do this , maybe purse it as a separate milestone **(Build Later)**
- Explore moving protocolFee to it's own contract? Eventually yes. Right now having it just for one / two variables is an overkill  **(Build Later)**

**RoundImplementation**
- [x] Adding RoundFee functionality (roundFeePercentage + roundFeeAddress)
- [x] Adding withdraw function to withdraw all tokens except round token
- Exploring using attestator contract (storage on a different contract) https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-periphery/contracts/universal/op-nft/AttestationStation.sol **Debate Later**

**IPayoutStrategy**
- [x] Do we need time lock and if yes why 60 ? **set timelock == 0**
- Checkout Aragon + Manifold + Zora for security vulnerabilities
- How are we batching + where are we storing the merkle proof
  - Native/ ERC-20 token: 200 txns. **Will deal with at UI layer**. 
  - **DO SPIKE TO SEE COST FOR MERKLE DISTRIBUTE** 
- Make payout function public **Debate later** 
- Where are we storing is KYC-ed info on the merkle proof   // Discuss 

**MerkleContract**
- Add a guard to ensure if round is for X tokens , only a max of X tokens can be paid out by the payout contract (to ensure the protocol fee is paid in full and bad actors don't fund the payout contract directly ) **DEBATE LATER**
```
    // TODO: create a var allowedFundsLeft
    // TODO: every time it is invoked -> reduce allowedFundsLeft - amount
    // check if allowedFundsLeft is >= 0
```

**Comment by vacekj** [comment]
Closing as stale
